acoular/spectra.py

```diff
@@ -77,23 +78,22 @@ class BaseSpectra(HasPrivateTraits):
     #:   * 'Hamming'
     #:   * 'Bartlett'
     #:   * 'Blackman'
-    window = Trait(
-        'Rectangular',
+    window = Map(
         {'Rectangular': ones, 'Hanning': hanning, 'Hamming': hamming, 'Bartlett': bartlett, 'Blackman': blackman},
+        default_value='Rectangular',
         desc='type of window for FFT',
     )
 
     #: Overlap factor for averaging: 'None'(default), '50%', '75%', '87.5%'.
-    overlap = Trait('None', {'None': 1, '50%': 2, '75%': 4, '87.5%': 8}, desc='overlap of FFT blocks')
+    overlap = Map({'None': 1, '50%': 2, '75%': 4, '87.5%': 8}, default_value='None', desc='overlap of FFT blocks')
 
     #: FFT block size, one of: 128, 256, 512, 1024, 2048 ... 65536,
     #: defaults to 1024.
-    block_size = Trait(
+    block_size = Enum(
         1024,
         128,
         256,
         512,
-        1024,
```

Comment:
was this intentional?

Reply:
yes, because the first value is also the default